app/components/Header/index.tsx

```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
 import Link from 'next/link';
-import styled from 'styled-components';
+const styled = require('styled-components').default;
```

Comment:
I'm not sure `const styled = require('styled-components').default` is a good approach. Why exactly `import styled from 'styled-components'` doesn't work?

Reply:
Looks like it has to do with this the issue 
https://github.com/styled-components/babel-plugin-styled-components/issues/41#issuecomment-297535321
When we use `import styled from 'styled-components'`, it gets transformed to `styled_components_1.default` by typescript which is not recognized as styled component helper by babel-plugin-styled-components.
I believe, this issue was present before, but wasn't obvious, because react 15 just replaced the mismatched server-generated markup. 
As long as react 16 doesn't do this anymore, it becomes evident in development mode. When a page gets reloaded, the styles don't apply to the components because of className mismatch.